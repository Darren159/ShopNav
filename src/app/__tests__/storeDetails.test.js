import { render } from "@testing-library/react-native";
import { Alert } from "react-native";
import { Router } from "expo-router";
import StoreDetails from "../storeDetails";

jest.mock("../../services/fetchPlaceId");
jest.mock("../../services/fetchPlaceDetails");
jest.spyOn(Alert, "alert");

describe("StoreDetails", () => {
  it("renders correctly", () => {
    const { toJSON } = render(
      <Router
        initialPath="/"
        initialParams={{
          locName: "test",
          promoInfo: "promo",
          storeName: "testStore",
        }}
      >
        <StoreDetails />
      </Router>
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
