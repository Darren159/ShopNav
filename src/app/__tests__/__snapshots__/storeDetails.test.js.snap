// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoreDetails renders correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "flex": 1,
      "marginHorizontal": 20,
    }
  }
>
  <ActivityIndicator
    color="#5500dc"
    size="large"
    style={
      {
        "flex": 1,
      }
    }
    testID="loading"
  />
</RCTSafeAreaView>
`;
